File: PROJECT/WEB/CODE/src/lib/storage.js
// -- IMPORTS

import { PUBLIC_FLUENT_PROJECT_STORAGE_URL } from '$env/static/public';

// -- FUNCTIONS

export function getStorageFilePath(
    filePath
    )
{
    if ( filePath.startsWith( '/upload/' ) )
    {
        return PUBLIC_FLUENT_PROJECT_STORAGE_URL + filePath;
    }
    else
    {
        return filePath;
    }
}

// ~~

export function getStorageImagePath(
    imagePath,
    imageWidth
    )
{
    if ( !isNaN( imageWidth ) )
    {
        if ( imagePath.endsWith( '.avif' ) )
        {
            imagePath += '.' + imageWidth + '.avif';
        }
        else if ( imagePath.endsWith( '.jpg' ) )
        {
            imagePath += '.' + imageWidth + '.jpg';
        }
        else if ( imagePath.endsWith( '.png' ) )
        {
            imagePath += '.' + imageWidth + '.png';
        }
    }

    return getStorageFilePath( imagePath );
}

File: PROJECT/MOBILE/CODE/lib/storage/storage.dart
// -- IMPORTS

import 'constant.dart';

// -- FUNCTIONS

String getStorageFilePath(
    String filePath
    )
{
    if ( filePath.startsWith( '/upload/' ) )
    {
        return storageUrl + filePath;
    }
    else
    {
        return filePath;
    }
}

// ~~

String getStorageImagePath(
    String imagePath,
    int imageWidth
    )
{
    if ( imagePath.endsWith( '.jpg' ) )
    {
        imagePath = '$imagePath.$imageWidth.jpg';
    }
    else if ( imagePath.endsWith( '.png' ) )
    {
        imagePath = '$imagePath.$imageWidth.png';
    }

    return getStorageFilePath( imagePath );
}

